factories/esSearcherFactory.js

```diff
@@ -169,7 +169,7 @@
 
               }
             }
-            else if (msg.status === -1) {
+            else if (msg.status === -1 || msg.status === 0) {
```

Comment:
tests were failing, and it seems like the although you're passing `-1` in the `response` method in the test, the actual number that was being received by this method is a `0`.

and honestly, I didn't want to spend too much time trying to figure out since it had nothing to do with the change Im making but I wanted the tests to pass. 

from first glance it seemed like it's an un-handled situation and could be dealt with the same way as a `-1` so I went with it. 

would be curious if those tests were passing for you, and why they wouldn't for me.
also seems like they might not be running on Travis CI?